gcloud-java-dns/src/test/java/com/google/cloud/dns/it/ITDnsTest.java

```diff
@@ -962,4 +965,26 @@ public void testListDnsRecords() {
       clear();
     }
   }
+
+  @Test
+  public void testListZoneBatch() {
+    DNS.create(ZONE1);
+    DNS.create(ZONE_EMPTY_DESCRIPTION);
+    try {
+      DnsBatch batch = DNS.batch();
+      DnsBatchResult<Page<Zone>> batchResult = batch.listZones();
+      batch.submit();
+      assertTrue(batchResult.submitted());
+      Iterator<Zone> iteratorBatch = batchResult.get().iterateAll();
+      Iterator<Zone> iteratorList = DNS.listZones().iterateAll();
+      while (iteratorBatch.hasNext()) {
+        assertTrue(Iterators.contains(iteratorList, iteratorBatch.next()));
```

Comment:
Looking at this test in isolation, I think this test could pass even if batch.listZones() were coded incorrectly (since we don't check the opposite as well, that iteratorBatch doesn't contain everything in iteratorList).  In reality, I think the tests you wrote in DnsBatchTest would catch that kind of error.  But if you want to add the check for the converse as well, it would make this test slightly stronger in my mind.